src/renderer/atlas/AtlasEngine.cpp

```diff
@@ -937,6 +937,21 @@ void AtlasEngine::_recreateSizeDependentResources()
         _api.glyphOffsets = Buffer<DWRITE_GLYPH_OFFSET>{ projectedGlyphSize };
     }
 
+    // Initialize cellGlyphMapping with valid data (whitespace), so that it can be
+    // safely used by the TileHashMap refresh logic via makeNewest() in StartPaint().
+    {
+        u16x2* coords;
+        AtlasKey key{ { .cellCount = 1 }, 1, L" " };
+        AtlasValue value{ CellFlags::None, 1, &coords };
+
+        coords[0] = _r.tileAllocator.allocate(_r.glyphs);
+
+        const auto it = _r.glyphs.insert(std::move(key), std::move(value));
+        _r.glyphQueue.emplace_back(it);
+
+        std::ranges::fill(_r.cellGlyphMapping, it);
+    }

```

Comment:
The crash occurs when the viewport size changes causing the recreation of the `cellGlyphMapping` array and the first paint includes non-single-width line rendition rows. In that case the `cellGlyphMapping` array will have items that are still default-constructed pointers (null pointers in debug, uninitialized data in release mode), leading to a crash.